### PR TITLE
build: set Python semverRange: '>=2.7.0 <3.8.0'

### PR DESCRIPTION
**Pull request by cclauss**
##### Checklist

- [ ] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change

Based on the discussion #1811 let's try really running the tests on Python 3.  This currently breaks that macOS builds.

### CODE REVIEW

File: lib/configure.js [hunk 1]
```diff
@@ -362,7 +362,7 @@ PythonFinder.prototype = {
   log: logWithPrefix(log, 'find Python'),
   argsExecutable: ['-c', 'import sys; print(sys.executable);'],
   argsVersion: ['-c', 'import sys; print("%s.%s.%s" % sys.version_info[:3]);'],
-  semverRange: '>=2.6.0 <3.0.0',
+  semverRange: '>=2.7.0 <3.8.0',
```
Comment (richardlau) [inline lib/configure.js]: Do we need an upper bound on the semver check anymore? It was there before to exclude Python 3.

**Comment by cclauss** [comment]
These tests would be more helpful if they printed __argsVersion__ on failure but I am not sure how to add that functionality.

**Comment by cclauss** [comment]
Given the "purely stylistic" changes in #1794, should I also whack our whole codebase thru [python/black](https://github.com/python/black)?

File: lib/configure.js (hunk 1 above)
Comment (cclauss) [reply lib/configure.js]: Now it is __semverRange: '>=2.7.0'__

File: .travis.yml [hunk 2]
```diff
@@ -29,6 +29,7 @@ matrix:
       python: 3.7
       before_install: nvm install 12
   allow_failures:
+    - os: osx
```
Comment (sam-github) [inline .travis.yml]: why?

File: lib/configure.js [hunk 3]
```diff
@@ -383,9 +383,9 @@ function PythonFinder (configPython, callback) {
 
 PythonFinder.prototype = {
   log: logWithPrefix(log, 'find Python'),
-  argsExecutable: [ '-c', 'import sys; print(sys.executable);' ],
-  argsVersion: [ '-c', 'import sys; print("%s.%s.%s" % sys.version_info[:3]);' ],
-  semverRange: '>=2.6.0 <3.0.0',
+  argsExecutable: [ '-c', 'import sys; print(sys.executable)' ],
```
Comment (sam-github) [inline lib/configure.js]: is dropping the `;` cosmetic, or necessary? If cosmetic, I'd prefer to leave as is to not mix cosmetic and functional changes.

**Comment by sam-github** [comment]
other than small comments, LGTM

**Comment by cclauss** [comment]
Closing in favor of #1815

**Pull request closed without merging**